### PR TITLE
Create a docker build script

### PR DESCRIPTION
**Pull request by erubboli**
This script build and pushes the docker images, it retrieve the version from the Cargo file and can be invoked by make as `build_docker_images`.

### CODE REVIEW

**Review by alfiedotwtf — APPROVED** [review]

**Review by muursh — APPROVED** [review]
I've half tested this. I have no issues with what I see.

File: build-tools/docker/build.py [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+    command = f"docker build -t {image_name}:{version} -f {dockerfile_path} ."
+
+    # Run the command
+    process = subprocess.Popen(command, shell=True)
+    output, error = process.communicate()
+
+    if error:
+        print(f"Error occurred: {error}")
+    else:
+        print(f"Built {image_name}:{version} successfully.")
```
Comment (ImplOfAnImpl) [inline build-tools/docker/build.py:33]: Shouldn't we stop the script if `docker build` fails? (i.e. why not just use `check_call` here like in other places?)

File: build-tools/docker/Dockerfile.builder [hunk 2]
```diff
@@ -0,0 +1,8 @@
+FROM rust AS builder
+
+WORKDIR /usr/src/
+
+COPY . .
+
+ARG NUM_JOBS=1
```
Comment (ImplOfAnImpl) [inline build-tools/docker/Dockerfile.builder:7]: At this moment it's impossible to specify this value when using `build.py`. I'd be nice if the latter had an additional parameter for this.

**Review by ImplOfAnImpl — APPROVED** [review]

File: Makefile [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+
+# python scripts are requiring `python3.11-venv` and `python3-pip` installed
+install-pip-dependencies:
+	python3 -m venv env && . env/bin/activate && python3 -m pip install --upgrade pip && python3 -m pip install toml
+
+build_docker_images: install-pip-dependencies
+	. env/bin/activate && python3 build-tools/docker/build.py
+
+push_docker_images: install-pip-dependencies
+	. env/bin/activate && python3 build-tools/docker/build.py --build=false --push --latest
```
[the remark this answers is not in the document]
Comment (erubboli) [reply Makefile:11]: I'll put the dependencies in the readme

**Pull request merged**